content/blog/deploying-mysql-schemas-using-dynamic-providers/index.md

```diff
@@ -0,0 +1,284 @@
+    publicly_accessible=True,
+    db_subnet_group_name=app_database_subnetgroup.id,
+    vpc_security_group_ids=[app_security_group.id])
+```
+
+Pulumi offers some additional tools to make handling MySQL easier.
+
+```python
+mysql_provider = mysql.Provider("mysql-provider",
+    endpoint=mysql_rds_server.endpoint,
+    username=admin_name,
+    password=admin_password)
+```
+
+An example database is initialized, and a user is created to manage it.
+
+```python
+mysql_database = mysql.Database("mysql-database",
+    name="votes-database",
+    opts=pulumi.ResourceOptions(provider=mysql_provider))
+
+mysql_user = mysql.User("mysql-standard-user",
+    user=user_name,
+    host="example.com",
+    plaintext_password=user_password,
+    opts=pulumi.ResourceOptions(provider=mysql_provider))
+```
+
+The user only needs the `SELECT` and `UPDATE` permissions to function.
+
+```python
+mysql_access_grant = mysql.Grant("mysql-access-grant",
+    user=mysql_user.user,
+    host=mysql_user.host,
+    database=mysql_database.name,
+    privileges= ["SELECT", "UPDATE"],
+    opts=pulumi.ResourceOptions(provider=mysql_provider))
+```
+
+Now, we begin to use the Dynamic Provider we created. The provider takes in a `creation_script` as a parameter, connects to our MySQL server, and runs it during deployment.
+
+```python
+creation_script = """
+    CREATE TABLE votesTable (
+        choice_id int(10) NOT NULL AUTO_INCREMENT,
+        vote_count int(10) NOT NULL,
+        PRIMARY KEY (choice_id)
+    ) ENGINE=InnoDB;
+    INSERT INTO votesTable(choice_id, vote_count) VALUES (0,0);
+    INSERT INTO votesTable(choice_id, vote_count) VALUES (1,0);
+    """
+```
+
+The provider also takes in a `deletion_script`, which it uses to undo all of its actions when the resource needs to be deleted.
+
+```python
+deletion_script = "DROP TABLE votesTable"
+```
+
+We can now create our resource. It behaves the same way as any other Pulumi resource, but takes its arguments as a `SchemaInputs` object.
+
+```python
+mysql_votes_table = Schema(name="mysql_votes_table",
+    args=SchemaInputs(admin_name, admin_password, mysql_rds_server.address, mysql_database.name, creation_script, deletion_script))
+```
+
+We can export the ID of our new resource, and view it when it is deployed.
+
+```python
+pulumi.export("dynamic-resource-id",mysql_votes_table.id)
+```
+
+In this example, I showed how straightforward it is to convert infrastructure code defined in TypeScript to one defined in Python. Teams often choose to standardize on one language and Pulumi expands that choice, allowing them to standardize on one language for the frontend, the backend, and the infrastructure.
+
+In this example, I showed how straightforward it is to write additional code for Pulumi and expand its functionality even further. Dynamic Providers and resources allow for excellent flexibility in cloud architecture design, and help break down barriers that would otherwise be difficult to overcome.
```

Comment:
```suggestion
In this example, I showed how straightforward it is to write additional code for Pulumi and expand its functionality even further. Dynamic Providers enable excellent flexibility in cloud architecture design, and help break down barriers that would otherwise be challenging to overcome.
```